.github/workflows/conformance-ingress.yaml

```diff
@@ -242,8 +242,25 @@ jobs:
       - name: Run Sanity check (external)
         timeout-minutes: 5
         run: |
+          set +e
           lb=$(kubectl get ingress basic-ingress -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
-          curl -s -v --connect-timeout 5 --max-time 20 --retry 3 --fail -- http://"$lb"
+          
+          # It may take some time after the service gets created for it to handle traffic.
+          # An un-ready service will return an ICMP so we retry a few times since the --retry
+          # flag only retries on connection timeouts.
+          curl -s -v --connect-timeout 5 --max-time 20 --fail -- http://"$lb"
+          if [ $? -ne 0 ]; then
+            sleep 5s
```

Comment:
We want to avoid sleeps in tests, and this could be a loop with a counter rather than nested, but I think both are _okay_ here.

Reply:
FWIW the danger is that if we have a bug that causes LB decisions to be flaky or inconsistent rather than moving from reliably not working to reliably working, then this style of testing will hide such flaky production behaviour.